Tests/SonarScanner.MSBuild.Shim.Test/PathHelperTests.cs

```diff
@@ -29,164 +30,162 @@ namespace SonarScanner.MSBuild.Shim.Test
-            PathHelper.IsInDirectory(new FileInfo("C:\\Src\\File.cs"), new DirectoryInfo("C:/Src")).Should().BeTrue();
-            PathHelper.IsInDirectory(new FileInfo("~Foo\\File.cs"), new DirectoryInfo("~Foo")).Should().BeTrue();
-            PathHelper.IsInDirectory(new FileInfo("C:\\Src\\Bar\\..\\File.cs"), new DirectoryInfo("C:\\Src")).Should().BeTrue();
-            PathHelper.IsInDirectory(new FileInfo("C:\\Src\\File.cs"), new DirectoryInfo("C:\\Src\\Bar\\..")).Should().BeTrue();
-
-            PathHelper.IsInDirectory(new FileInfo("C:\\SrcFile.cs"), new DirectoryInfo("C:\\Src")).Should().BeFalse();
-            PathHelper.IsInDirectory(new FileInfo("C:\\Src\\File.cs"), new DirectoryInfo("C:\\Src\\Bar")).Should().BeFalse();
+            PathHelper.IsInDirectory(new FileInfo(@"C:\Src\File.cs"), new DirectoryInfo(@"C:\Src\")).Should().BeTrue();
+            PathHelper.IsInDirectory(new FileInfo(@"C:\Src\File.cs"), new DirectoryInfo(@"C:\Src")).Should().BeTrue();
+            PathHelper.IsInDirectory(new FileInfo(@"C:\Src\File.cs"), new DirectoryInfo(@"C:\")).Should().BeTrue();
+            PathHelper.IsInDirectory(new FileInfo(@"C:\SRC\FILE.CS"), new DirectoryInfo(@"C:\src")).Should().BeTrue();
+            PathHelper.IsInDirectory(new FileInfo(@"C:/Src/File.cs"), new DirectoryInfo(@"C:/Src")).Should().BeTrue();
+            PathHelper.IsInDirectory(new FileInfo(@"C:\Src\File.cs"), new DirectoryInfo(@"C:/Src")).Should().BeTrue();
+            PathHelper.IsInDirectory(new FileInfo(@"~Foo\File.cs"), new DirectoryInfo("~Foo")).Should().BeTrue();
+            PathHelper.IsInDirectory(new FileInfo(@"C:\Src\Bar\..\File.cs"), new DirectoryInfo(@"C:\Src")).Should().BeTrue();
+            PathHelper.IsInDirectory(new FileInfo(@"C:\Src\File.cs"), new DirectoryInfo(@"C:\Src\Bar\..")).Should().BeTrue();
+            // False
+            PathHelper.IsInDirectory(new FileInfo(@"C:\SrcFile.cs"), new DirectoryInfo(@"C:\Src")).Should().BeFalse();
+            PathHelper.IsInDirectory(new FileInfo(@"C:\Src\File.cs"), new DirectoryInfo(@"C:\Src\Bar")).Should().BeFalse();
         }
 
         [TestMethod]
-        public void GetCommonRoot_WhenNull_ReturnsNull()
-        {
-            // Arrange
-            // Act
-            var result = PathHelper.GetCommonRoot(null);
+        public void BestCommonPrefix_WhenNull_ReturnsNull() =>
+            PathHelper.BestCommonPrefix(null).Should().BeNull();
 
-            // Assert
-            result.Should().BeNull();
-        }
+        [TestMethod]
+        public void BestCommonPrefix_WhenEmpty_ReturnsNull() =>
+            PathHelper.BestCommonPrefix(Enumerable.Empty<DirectoryInfo>()).Should().BeNull();
 
         [TestMethod]
-        public void GetCommonRoot_WhenNoCommonPath_ReturnsNull()
-        {
-            PathHelper.GetCommonRoot(new[]
+        public void BestCommonPrefix_WhenNoCommonPath_ReturnsNull() =>
```

Comment:
this is the same as `BestCommonPrefix_WhenNoCommonPath_SameCountInEachGroup_ReturnsNull()`, right?

seeing just "WhenNoCommonPath" here is confusing because it's actually "same count in each group"

maybe drop it or add the same _Simple and _Complex suffixes?